Error from server (AlreadyExists): project.project.openshift.io "test-project" already exists

failed with -m tier1 run

```
        #log.debug(f"Command output: {r.stdout.decode()}")
        if r.stderr and not r.returncode:
            log.warning(f"Command warning:: {r.stderr.decode()}")
        if r.returncode:
            raise CommandFailed(
>               f"Error during execution of command: {cmd}."
                f"\nError is {r.stderr.decode()}"
            )
E           ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: ['oc', 'new-project', 'test-project'].
E           Error is Error from server (AlreadyExists): project.project.openshift.io "test-project" already exists
```

and the same test script before failed with

```
tests/manage/test_pvc_concurrent_deletion_creation.py:30: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <class 'tests.manage.test_pvc_concurrent_deletion_creation.TestMultiplePvcConcurrentDeletionCreation'>

    def teardown(self):
        """
        Delete PVCs
        Delete project
        """
        # Delete newly created PVCs
        command = (
            f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
            f'{self.pvc_base_name_new}$i -n {TEST_PROJECT};done'
        )
        proc = run_async(command)
        assert proc, (
            f'Failed to execute command for deleting {self.number_of_pvc} PVCs'
        )
    
        # Verify command to delete PVCs
        ret, _, _ = proc.async_communicate()
>       assert not ret, 'Deletion of newly created PVCs failed'
E       AssertionError: Deletion of newly created PVCs failed
E       assert not 1
```

--- a/ocs_ci/ocs/resources/pvc.py
+++ b/ocs_ci/ocs/resources/pvc.py
@@ -74,6 +74,9 @@ def delete_all_pvcs(namespace=None):
     """
     Deletes all pvc in namespace
 
+    Args:
+        namespace (str): Name of namespace
+
     Returns:
         bool: True if deletion is successful
     """
@@ -93,6 +96,9 @@ def delete_all_pvcs(namespace=None):
 def get_all_pvcs(namespace=None):
     """
     Gets all pvc in given namespace
+
+    Args:
+        namespace (str): Name of namespace
 
     Returns:
          dict: Dict of all pvc in namespaces

--- a/ocs_ci/ocs/resources/pvc.py
+++ b/ocs_ci/ocs/resources/pvc.py
@@ -70,15 +70,17 @@ class PVC(OCS):
         return True
 
 
-def delete_all_pvcs():
+def delete_all_pvcs(namespace=None):
     """
     Deletes all pvc in namespace
 
     Returns:
         bool: True if deletion is successful
     """
+    if not namespace:
+        namespace = config.ENV_DATA['cluster_namespace']
     ocp_pvc_obj = OCP(
-        kind=constants.PVC, namespace=config.ENV_DATA['cluster_namespace']
+        kind=constants.PVC, namespace=namespace
     )
     ocp_pvc_list = get_all_pvcs()
     pvc_list = ocp_pvc_list['items']
@@ -88,16 +90,17 @@ def delete_all_pvcs():
     return True
 
 
-def get_all_pvcs():
+def get_all_pvcs(namespace=None):
     """
     Gets all pvc in given namespace
 
     Returns:
          dict: Dict of all pvc in namespaces
     """
-
+    if not namespace:
+        namespace = config.ENV_DATA['cluster_namespace']
     ocp_pvc_obj = OCP(
-        kind=constants.PVC, namespace=config.ENV_DATA['cluster_namespace']
+        kind=constants.PVC, namespace=namespace
     )
     out = ocp_pvc_obj.get()
     return out

--- a/ocs_ci/ocs/resources/pvc.py
+++ b/ocs_ci/ocs/resources/pvc.py
@@ -85,7 +85,7 @@ def delete_all_pvcs(namespace=None):
     ocp_pvc_obj = OCP(
         kind=constants.PVC, namespace=namespace
     )
-    ocp_pvc_list = get_all_pvcs()
+    ocp_pvc_list = get_all_pvcs(namespace=namespace)
     pvc_list = ocp_pvc_list['items']
     for item in pvc_list:
         ocp_pvc_obj.delete(resource_name=item.get('metadata').get('name'))

--- a/tests/manage/test_pvc_concurrent_deletion_creation.py
+++ b/tests/manage/test_pvc_concurrent_deletion_creation.py
@@ -8,7 +8,7 @@ from ocs_ci.ocs import constants, ocp, exceptions
 from ocs_ci.utility.utils import run_async
 from ocs_ci.framework.testlib import tier1, ManageTest
 from ocs_ci.ocs.resources.pod import get_ceph_tools_pod
-from ocs_ci.ocs.resources.pvc import create_multiple_pvc
+from ocs_ci.ocs.resources.pvc import create_multiple_pvc, delete_all_pvcs
 from tests.fixtures import (
     create_rbd_storageclass, create_ceph_block_pool, create_rbd_secret
 )
@@ -69,18 +69,7 @@ def teardown(self):
     Delete project
     """
     # Delete newly created PVCs
-    command = (
-        f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
-        f'{self.pvc_base_name_new}$i -n {TEST_PROJECT};done'
-    )
-    proc = run_async(command)
-    assert proc, (
-        f'Failed to execute command for deleting {self.number_of_pvc} PVCs'
-    )
-
-    # Verify command to delete PVCs
-    ret, _, _ = proc.async_communicate()
-    assert not ret, 'Deletion of newly created PVCs failed'
+    assert delete_all_pvcs(namespace=TEST_PROJECT), 'Failed to delete PVCs'
     log.info(f'Newly created {self.number_of_pvc} PVCs are now deleted.')
 
     # Switch to default project

--- a/tests/manage/test_pvc_concurrent_deletion_creation.py
+++ b/tests/manage/test_pvc_concurrent_deletion_creation.py
@@ -13,10 +13,9 @@ from tests.fixtures import (
     create_rbd_storageclass, create_ceph_block_pool, create_rbd_secret
 )
 from ocs_ci.utility.templating import load_yaml_to_dict
+from tests.helpers import create_unique_resource_name
 
 log = logging.getLogger(__name__)
-TEST_PROJECT = 'test-project'
-PROJECT = ocp.OCP(kind='Project', namespace=TEST_PROJECT)
 
 
 @pytest.fixture()
@@ -38,13 +37,15 @@ def setup(self):
     Create PVCs
     """
     # Create new project
-    assert PROJECT.new_project(TEST_PROJECT), (
-        f'Failed to create new project {TEST_PROJECT}'
+    self.namespace = create_unique_resource_name('test', 'namespace')
+    self.project_obj = ocp.OCP(kind='Project', namespace=self.namespace)
+    assert self.project_obj.new_project(self.namespace), (
+        f'Failed to create new project {self.namespace}'
     )
 
     # Parameters for PVC yaml as dict
     pvc_data = load_yaml_to_dict(constants.CSI_PVC_YAML)
-    pvc_data['metadata']['namespace'] = TEST_PROJECT
+    pvc_data['metadata']['namespace'] = self.namespace
     pvc_data['spec']['storageClassName'] = self.sc_obj.name
     pvc_data['metadata']['name'] = self.pvc_base_name
 
@@ -69,7 +70,7 @@ def teardown(self):
     Delete project
     """
     # Delete newly created PVCs
-    assert delete_all_pvcs(namespace=TEST_PROJECT), 'Failed to delete PVCs'
+    assert delete_all_pvcs(namespace=self.namespace), 'Failed to delete PVCs'
     log.info(f'Newly created {self.number_of_pvc} PVCs are now deleted.')
 
     # Switch to default project
@@ -77,7 +78,7 @@ def teardown(self):
     assert ret, 'Failed to switch to default rook cluster project'
 
     # Delete project created for the test case
-    PROJECT.delete(resource_name=TEST_PROJECT)
+    self.project_obj.delete(resource_name=self.namespace)
 
 
 @tier1
@@ -105,7 +106,7 @@ class TestMultiplePvcConcurrentDeletionCreation(ManageTest):
         # Start deleting 100 PVCs
         command = (
             f'for i in `seq 1 {self.number_of_pvc}`;do oc delete pvc '
-            f'{self.pvc_base_name}$i -n {TEST_PROJECT};done'
+            f'{self.pvc_base_name}$i -n {self.namespace};done'
         )
         proc = run_async(command)
         assert proc, (
@@ -115,7 +116,7 @@ class TestMultiplePvcConcurrentDeletionCreation(ManageTest):
         # Create 100 new PVCs
         # Parameters for PVC yaml as dict
         pvc_data = load_yaml_to_dict(constants.CSI_PVC_YAML)
-        pvc_data['metadata']['namespace'] = TEST_PROJECT
+        pvc_data['metadata']['namespace'] = self.namespace
         pvc_data['spec']['storageClassName'] = self.sc_obj.name
         pvc_data['metadata']['name'] = self.pvc_base_name_new
 

